okhttp/src/main/kotlin/okhttp3/internal/connection/PoolConnectionUser.kt

```diff
@@ -0,0 +1,105 @@
+package okhttp3.internal.connection
+
+import java.io.IOException
+import java.net.InetAddress
+import java.net.Proxy
+import java.net.Socket
+import okhttp3.Connection
+import okhttp3.Handshake
+import okhttp3.HttpUrl
+import okhttp3.Protocol
+import okhttp3.Route
+
+/**
+ * A user that is a connection pool creating connections in the background
+ * without an intent to immediately use them.
+ *
+ * Most of this class's functionality is currently stubbed out, but will be revisited in the future.
+ */
+class PoolConnectionUser : ConnectionUser {
```

Comment:
This class is currently a giant no-op. We intend to think more carefully about what the "right" thing to do here is, but first we want to make sure this feature works at all.